### ISSUE TITLE
`info()` result `auto_compaction` property

### ISSUE THREAD

**Issue by AGBrown** [AGBrown/pouchdb.d.ts#6]
#### Background

In the [api docs](http://pouchdb.com/api.html#database_information) it lists the [response object](https://github.com/AGBrown/pouchdb.d.ts/blob/56a04ad1/pouchdb.d.ts#L825-L834) as (in .ts form):

``` javascript
/** Response object for `info()` */
interface Response {
    /** the name of the database, and also the unique identifier for the database */
    db_name: string;
    /** the total number of non-deleted documents in the database */
    doc_count: number;
    /** the sequence number of the database. It starts at 0 and gets incremented 
      * every time a document is added or modified. */
    update_seq: number
}
```

it also states:

> There are also some details you can use for debugging. These are unofficial and may change at any time

Thus far I have represented these as a [special case interface](https://github.com/AGBrown/pouchdb.d.ts/blob/56a04ad1/pouchdb.d.ts#L835-L843) for advanced use:

``` javascript
/** Response object for `info()` */
interface ResponseDebug extends Response {
    /** (IndexedDB) either `'base64'` or `'binary'` */
    idb_attachment_format: string;
    /** (WebSQL) true if the SQLite Plugin is being used */
    sqlite_plugin: string;
    /** (WebSQL) either `'UTF-8'` or `'UTF-16'`. */
    websql_encoding: string
}
```

the [result of](https://github.com/AGBrown/pouchdb.d.ts/blob/56a04ad1/pouchdb.d.ts#L852) `info()` can therefore be "normally" treated as a `Response` object, and optionally cast to a `ResponseDebug` to access the extra properties in advanced usage.
#### Question

In the tests `'db.info should give auto_compaction = ...` (ref. pouchdb/pouchdb#2744)' in [test.basics.js](https://github.com/pouchdb/pouchdb/blob/9890648/tests/integration/test.basics.js#L758-L771) there is also an `auto_compaction` property.

Should this be considered:
1. "debug info and subject to change" (i.e. for advanced use only and put on `ResponseDebug`); or
2. a property that is commonly used by any developer and therefore put on `Response`?

**Comment by nolanlawson** [comment]
Yeah, I think we forgot to document the `auto_compaction` property on the info object.

In general, you will have a tough time formalizing those keys, because they vary wildly among different *ouch implementations (check out the difference between PouchDB Server, CouchDB 1.x, CouchDB 2.x, Cloudant, and Couchbase Sync Gateway sometime)...

I dunno what the standard is for such cases in TypeScript, but maybe you could just have an object that's a `Map<String,Object>` called `params` or something?

**Comment by AGBrown** [comment]
@nolanlawson I don't actually know if I can do that, but I will look into it.

It could also be the case that we build up a set of different info response objects that have keys based on the underlying, and then the end user can cast to the right object.

As with all ts I could just return the info as `any` - it would all still be valid, but adding the most common properties to the returned object's interface helps people using the d.ts feel their way through the code by following the intellisense.

I'll leave the `auto_compaction` on the `ResponseDebug` interface as it doesn't seem to be commonly used, also as per IRC earlier @marten-de-vries and yourself seem to say the same thing:

> <kbd>[19:00:31] <Freenode:#pouchdb:commandoline></kbd> I consider `auto_compaction` to be a `ResponseDebug` property - simply because CouchDB doesn't support it and it's not _specifically_ listed in our documentation.

**Comment by AGBrown** [comment]
Closed for now - [e3d87cac](https://github.com/AGBrown/pouchdb.d.ts/blob/e3d87cac/pouchdb.d.ts#L844) seems to match what @nolanlawson  and @marten-de-vries are saying, and the d.ts now [allows the tests to be written](https://github.com/AGBrown/pouchdb.d.ts/blob/e3d87cac/tests/integration/test.basics.ts#L787-L800).
